### ISSUE TITLE
proposal: expand `prefer_null_aware_operators` to cover simple `if` clauses

### ISSUE THREAD

**Issue by LongCatIsLooong** [dart-lang/linter#4025]
## Description

If `child` is not a local variable, prefer `child?.detach()` over 
```dart
if (child != null) {
  child!.detach();
}
```

## Details

**PREFER** using null aware operators instead of null checks in conditional expressions and simple if clauses.

## Kind

style advice. `a?.doStuff()` is more idiomatic.

## Bad Examples

```dart
if (a != null) {
  a!.doStuff();
}
```
where `a` can't be promoted to non-null. 

## Good Examples 

```dart
a?.doStuff();
```
## Discussion

The lint should probably only fire when the if clause takes the exact form of
```dart
if (a != null) { 
  a!.doStuff(b);
}
```
to ensure nothing else in the `if` clause (other than `a` and `b`, given that `b` won't be evaluated in `a?.doStuff(b)`, if `a == null`) depends on `a != null`,  to avoid false positives.  

### Discussion checklist

- [ ] List any existing rules this proposal modifies, complements, overlaps or conflicts with.
- [ ] List any relevant issues (reported here, the [SDK Tracker], or elsewhere).
- [ ] If there's any prior art (e.g., in other linters), please add references here.
- [ ] If this proposal corresponds to [Effective Dart] or [Flutter Style Guide] advice, please call it out. (If there isn't any corresponding advice, should there be?)
- [ ] If this proposal is motivated by real-world examples, please provide as many details as you can.  Demonstrating potential impact is _especially valuable_.

[Writing Lints]: https://github.com/dart-lang/linter/blob/main/doc/writing-lints.md
[Effective Dart]: https://dart.dev/guides/language/effective-dart
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[SDK Tracker]: https://github.com/dart-lang/sdk/issues

**Comment by oprypin** [comment]
Isn't this already available as [prefer_null_aware_method_calls](https://dart-lang.github.io/linter/lints/prefer_null_aware_method_calls.html) ?

**Comment by srawlins** [comment]
I believe so.
